File: app/src/main/java.backup/com/lumiyaviewer/lumiya/slproto/messages/CompletePingCheck.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.google.common.primitives.UnsignedBytes;
import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;

public class CompletePingCheck extends SLMessage {
    public PingID PingID_Field = new PingID();

    public static class PingID {
        public int PingID;
    }

    public CompletePingCheck() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 2;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleCompletePingCheck(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.put((byte) 2);
        packByte(byteBuffer, (byte) this.PingID_Field.PingID);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.PingID_Field.PingID = unpackByte(byteBuffer) & UnsignedBytes.MAX_VALUE;
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/CompletePingCheck.kt
package com.linkpoint.slproto.messages

import com.google.common.primitives.UnsignedBytes
import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class CompletePingCheck : SLMessage() {
    val PingID_Field = PingID()

    class PingID {
        var PingID: Int = 0
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 2
    }

    override fun Handle(handler: SLMessageHandler) {
        handler.HandleCompletePingCheck(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {
        buffer.put(2.toByte())
        packByte(buffer, PingID_Field.PingID.toByte())
    }

    override fun UnpackPayload(buffer: ByteBuffer) {
        PingID_Field.PingID = unpackByte(buffer).toInt() and UnsignedBytes.MAX_VALUE.toInt()
    }
}